### PR TITLE
Automatize "[4502] Event - follow-up of actions: attach documents at the level of the action carried out"

### PR DESCRIPTION
**Pull request by mp-sgent**

### CODE REVIEW

**Comment by mp-sgent** [comment]
https://jenkins.sormas.netzlink.com/view/SORMAS-E2eTests/job/sormas-RunE2eTests-branch/1534/allure/#

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java [hunk 1]
```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
...
+        (String fileType) -> {
+          webDriverHelpers.sendFile(FILE_PICKER, userDirPath + "/uploads/" + fileType);
+          TimeUnit.SECONDS.sleep(2); // wait for upload file
+        });
+    When(
+        "I check if ([^\"]*) file is available in Event Action documents",
+        (String fileType) -> {
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              By.xpath(String.format(CASE_UPLOADED_TEST_FILE, fileType)), 5);
+          TimeUnit.SECONDS.sleep(2);
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:83]: why this sleep?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java [hunk 2]
```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
...
+        "I click on ([^\"]*) button from NEW DOCUMENT in Event Action tab",
+        (String buttonName) -> {
+          webDriverHelpers.clickOnWebElementBySelector(NEW_DOCUMENT_BUTTON);
+          webDriverHelpers.clickWebElementByText(START_DATA_IMPORT_BUTTON, buttonName);
+        });
+    When(
+        "I upload ([^\"]*) file to the Event Action",
+        (String fileType) -> {
+          webDriverHelpers.sendFile(FILE_PICKER, userDirPath + "/uploads/" + fileType);
+          TimeUnit.SECONDS.sleep(2); // wait for upload file
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:76]: the file might be autoamatically uploaded, not sure if sleep is needed

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java [hunk 3]
```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
...
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              By.xpath(String.format(CASE_UPLOADED_TEST_FILE, fileType)), 5);
+          TimeUnit.SECONDS.sleep(2);
+        });
+    When(
+        "I check if ([^\"]*) file is downloaded correctly from Event Action tab",
+        (String filename) -> {
+          String file = userDirPath + "/downloads/" + filename;
+          Path path = Paths.get(file);
+          softly.assertTrue(Files.exists(path));
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:90]: assert message missing

Comment (rdutu-vg) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:90]: assertHelpers.assertWithPoll(
              () ->
                  Assert.assertTrue(
                      Files.exists(path),
                      dictionaryName + " wasn't downloaded: " + path.toAbsolutePath()),
              30); 

And you can use this approach to wait after file to be downloaded and check it

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java [hunk 4]
```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
...
+          softly.assertAll();
+          Files.delete(path); // clean
+        });
+    When(
+        "I download last updated document file from Event Action tab",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(DOWNLOAD_LAST_UPDATED_DOCUMENT);
+          TimeUnit.SECONDS.sleep(3); // wait for download
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:98]: remove this

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java [hunk 5]
```diff
@@ -49,6 +62,55 @@ public EditActionSteps(WebDriverHelpers webDriverHelpers, BaseSteps baseSteps) {
...
+        });
+    When(
+        "I delete last uploaded document file from Event Action tab",
+        () -> {
+          webDriverHelpers.clickOnWebElementBySelector(DELETE_LAST_UPDATED_DOCUMENT);
+          webDriverHelpers.clickOnWebElementBySelector(ACTION_CONFIRM_POPUP_BUTTON);
+          TimeUnit.SECONDS.sleep(5); // wait for system reaction
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/actions/EditActionSteps.java:105]: remove this as long as we have the spinner loading handle

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**